pingcap/tiflash
# Add background gc check thread for DeltaTree storage

### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tics/issues/1507

Problem Summary: Current storage gc is just triggered by write. And when the gc process run, the old version data may not be outdated. We need background gc thread to trigger gc intermittently.

### What is changed and how it works?

Proposal: https://docs.google.com/document/d/1O2X1IXGTLz0raO6uHKxaIu1dupJtqTkpfSoUyy4mfK0/edit

What's Changed: Add a background gc thread and some statistical info for DTFile to help to determine whether gc is needed.

### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`:
- Need to cherry-pick to the release branch:

### Check List 

Tests 

- Unit test
- Manual test (add detailed scripts or steps below)
  1.  run bank test on the cluster and see whether there is gc triggered by backgroung gc check thread.
  2. upgrade and downgrade and verify the cluster work normally.

Side effects
- Performance regression
     - Consumes more CPU
     - Consumes more MEM


### Release note 

- Fix the problem that obsolete data may not be removed when there is no more write to the storage.
